### PR TITLE
[ModelZoo] unify with elastic dl model zoo

### PR DESCRIPTION
**Pull request by typhoonzero**
The previous change https://github.com/sql-machine-learning/models/pull/24 is not correct comparing to https://github.com/sql-machine-learning/elasticdl/blob/develop/model_zoo/cifar10_subclass/cifar10_subclass.py.

This PR changes how the customed model will be used in SQLFlow: 

1. The model name changed to the python module name, so that functional API could be used in SQLFlow  using the same code: see https://github.com/sql-machine-learning/models/pull/23/files
1. `optimizer`, `loss` function moved to module function, not a member of the model class to be the same as ElasticDL. And, if we put `optimizer`, `loss` function inside the model class, it will cause keras model prediction error: https://travis-ci.com/sql-machine-learning/sqlflow/jobs/254846188

### CODE REVIEW

File: sqlflow_models/__init__.py [hunk 1]
```diff
@@ -1,4 +1,8 @@
 from ._version import __version__
-from .dnnclassifier import DNNClassifier
-from .lstmclassifier import StackedBiLSTMClassifier
-from .deep_embedding_cluster import DeepEmbeddingClusterModel
+# from .dnnclassifier import DNNClassifier
+# from .lstmclassifier import StackedBiLSTMClassifier
+# from .deep_embedding_cluster import DeepEmbeddingClusterModel
```
Comment (terrytangyuan) [inline sqlflow_models/__init__.py]: Remove commented out imports

File: sqlflow_models/deep_embedding_cluster.py [hunk 2]
```diff
@@ -272,6 +268,48 @@ def display_model_info(self, verbose=0):
...
+        pretrain_initializer=pretrain_initializer,
+        pretrain_lr=pretrain_lr,
+        train_lr=train_lr,
+        train_max_iters=train_max_iters,
+        update_interval=update_interval,
+        tol=tol,
+        loss=loss)
+
+def optimizer():
+    global _train_lr
```
Comment (terrytangyuan) [inline sqlflow_models/deep_embedding_cluster.py:269]: Why are these globals needed?

File: sqlflow_models/dnnclassifier.py [hunk 3]
```diff
@@ -25,14 +25,17 @@ def call(self, inputs):
...
+    """Default optimizer name. Used in model.compile."""
+    return tf.keras.optimizers.Adagrad(lr=learning_rate)
 
-    def prepare_prediction_column(self, prediction):
-        """Return the class label of highest probability."""
-        return prediction.argmax(axis=-1)
\ No newline at end of file
+def loss():
+    """Default loss function. Used in model.compile."""
+    return 'sparse_categorical_crossentropy'
```
Comment (terrytangyuan) [inline sqlflow_models/dnnclassifier.py:34]: I am not sure if string representation would work in ElasticDL yet.

Comment (typhoonzero) [reply sqlflow_models/dnnclassifier.py:34]: Will add unit tests on models repo when merging ElasticDL `model_zoo/` dir.

File: sqlflow_models/deep_embedding_cluster.py (hunk 2 above)
Comment (typhoonzero) [reply sqlflow_models/deep_embedding_cluster.py:269]: `_train_lr` may be configured by SQL statements when initializing the model instance. e.g.

```
SELECT * FROM mytrain_table TO TRAIN sqlflow_models.DNNClassifier WITH train_lr=0.0001 ...
```

In the above SQL statement, `train_lr` will be passed to `DeepEmbeddingClusterModel` to configure the learning rate.

File: sqlflow_models/dnnclassifier.py (hunk 3 above)
Comment (terrytangyuan) [reply sqlflow_models/dnnclassifier.py:34]: Do you mean we'll test this later? I am quite sure this won't work in ElasticDL yet.

File: sqlflow_models/lstmclassifier.py [hunk 4]
```diff
@@ -10,6 +12,7 @@ def __init__(self, feature_columns, stack_units=[32], hidden_size=64, n_classes=
         :param n_classes: Target number of classes.
         :type n_classes: int.
         """
+        global _loss
```
Comment (terrytangyuan) [inline sqlflow_models/lstmclassifier.py:15]: Why are these `global` keywords needed?

File: sqlflow_models/dnnclassifier.py (hunk 3 above)
Comment (typhoonzero) [reply sqlflow_models/dnnclassifier.py:34]: Yep. Will fix this later in another PR.

File: sqlflow_models/lstmclassifier.py (hunk 4 above)
Comment (typhoonzero) [reply sqlflow_models/lstmclassifier.py:15]: Please, check out the above comment:

`_train_lr` may be configured by SQL statements when initializing the model instance. e.g.

```
SELECT * FROM mytrain_table TO TRAIN sqlflow_models.DNNClassifier WITH train_lr=0.0001 ...
```

In the above SQL statement, `train_lr` will be passed to `DeepEmbeddingClusterModel` to configure the learning rate.

**Review by terrytangyuan — APPROVED** [review]
